training/train.py

```diff
@@ -302,30 +311,38 @@ def create_git_pr(s3_client, model_version, recall_at_30):  # pragma: no cover
     prev_hyperparams = s3_client.read_json_file(k)
 
     # Convert the json description to string
-    description = json.dumps(prev_hyperparams).replace('"', '\\"')
+    description = 'Previous hyper parameters :: ' + json.dumps(prev_hyperparams).replace('"', '\\"')
 
     prev_recall = prev_hyperparams.get('recall_at_30', 0.55)
+    _logger.info('create_git_pr:: Prev => Model {}, Recall {}  Curr => Model {}, Recall {}'.format(
+        previous_version, prev_recall, model_version, recall_at_30))
     if recall_at_30 >= prev_recall:
         try:
-            # Invoke bash script to create a saas-analytics PR
-            t = subprocess.Popen(['sh', 'rudra/utils/github_helper.sh', 'f8a-hpf-insights.yaml',
-                                  'MODEL_VERSION', str(model_version), description],
-                                 shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            # Wait for the subprocess to get over
-            t.wait(60)
-            if t.returncode == 0:
-                _logger.info("Successfully created a PR")
+            # 1. Get the bash script from rudra to raise PR
+            exec_command(['wget', '-v',
+                          'https://raw.githubusercontent.com/fabric8-analytics/'
+                          'fabric8-analytics-rudra/master/rudra/utils/github_helper.sh',
+                          '-O', './github_helper.sh'], 60)
+
+            # 2. Provide execute permission to the script file.
+            exec_command(['chmod', '+x', './github_helper.sh'], 30)
+
+            # 3. Invoke bash script to create a saas-analytics PR
+            exec_command(['./github_helper.sh', 'f8a-pypi-insights.yaml', 'MODEL_VERSION',
+                          str(model_version), description], 60)
         except ValueError:
             _logger.error('ERROR - Wrong number of arguments passed to subprocess')
             raise ValueError
         except subprocess.TimeoutExpired as s:
```

Comment:
Does it not make sense to move the exception handling also to that method as the subprocess is being created there?

Reply:
Moved process related exception handling in the function and wrapped existing block with generic exception to abort at first failure.